portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -0,0 +1,68 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for listing comments. */
+@WebServlet("/list-comments")
+public class ListCommentsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Obtain # of comments to display from URL parameter input.
+    String numString = request.getParameter("requestComments");
+    int requestComments = getNumber(numString);
+
+    // Sort entities according to time stamp.
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);;
```

Comment:
Can you limit the number of results returned using the query rather than grabbing all the comments then limiting the size? https://cloud.google.com/appengine/docs/standard/java/datastore/retrieving-query-results#setting_limits

Reply:
Yes, that is a more efficient approach - saves time and space. Done.